### DISCUSSION TITLE
Error Message - "Unknown or unsupported version of Mirth Connect"

### DISCUSSION THREAD

**Discussion by Jaysarun** [nextgenhealthcare/connect#6049] [Q&A]
Hi All,

We are currently upgraded our Mirth connect from 3.8.1 to 4.4.1 due to the vulnerability issues. After the upgraded the mirth we did imported our channels back. but when we try to test the channel with messages it is throwing this error message as 

"Unknown or unsupported version of Mirth Connect"

![image](https://github.com/nextgenhealthcare/connect/assets/155644178/cc2f73e5-514e-4613-b83b-c4428d341461)

Please share your thoughts and suggestions to resolve this one.

Thanks,
Arunprasanna Jay S

**Comment by jonbartels** [comment]
This looks like code in your channels. The stack trace shows it is coming from a Javascript transformer in a channel.  That error string does not appear anywhere in the Mirth code: https://github.com/search?q=repo%3Anextgenhealthcare%2Fconnect+%22unknown+or+unsupported%22&type=code

If you need help fixing your channel code, you need to post the code. Please use formatting when you do!

**Reply by Jaysarun** [reply]
Thanks Jon, I will try to debug our javascript and let you know any help needed. Appreciate it!

**Reply by Jaysarun** [reply]
Hi Jon,
I am currently using this JavaScript code to setup the DB connection, whenever I try to send a message it got errored out and getting the same error message as "Unknown or unsupported version of Mirth Connect" But the same code was working fine in the Previous version 3.8.1 now not in 4.4.1. Could you please suggest any.

Please find the code below,

var sqlUtils =  new Packages.com.quantros.utils.SQLUtils();
var DBMS_URL="train";
var vErrors="";
/**************************** START FUNCTIONS ****************************************/

var checkManFields =  function(arrlabels,arr)
{
	var errorMsgs="";
	if (arr.length!=arrlabels.length)
	{
		errorMsgs+="Tahoe 16801 Labels and columns array must match";
	}
	for(var i=0; i<arr.length; i++)
	{
		if (arr[i]=="")
		{
			errorMsgs+="Tahoe 16801 Mandatory: " + arrlabels[i];	
		}		
	}

	return errorMsgs;
}

function executeUpdate(strSQL,inParamsArray,inParamsTypesArray)
{
	return sqlUtils.executeUpdate(DBMS_URL,strSQL,makeJavaArr(inParamsArray),makeJavaArr(inParamsTypesArray));
}

function executeSingleRowQuery(strSQL,inParamsArray,inParamsTypesArray)
{
	var s=sqlUtils.executeSingleRowQuery(DBMS_URL,strSQL,makeJavaArr(inParamsArray),makeJavaArr(inParamsTypesArray));
	var arr = s.split(",");
	return arr;
}


function executeSingleValQuery(strSQL,inParamsArray,inParamsTypesArray)
{
	return sqlUtils.executeSingleValQuery(DBMS_URL,strSQL,makeJavaArr(inParamsArray),makeJavaArr(inParamsTypesArray));
}

function makeJavaArr(jsArray)
{
	var ae = java.lang.reflect.Array.newInstance(java.lang.String, jsArray.length); 
	 for(var i=0; i<jsArray.length; i++)
	 {
	 	ae[i]=jsArray[i];
	 }
	 return ae;
}

function arrayToString(arr)

{
	var s="";
	for(var i=0; i<arr.length; i++)
	{
		s+= arr[i] +",";
	}
	return s;
}

function checkSize(elemLabel,elemValue,sizeSpec)
{
	if (elemValue.length > sizeSpec)
	{
		vErrors += elemLabel + "Tahoe 16801 Size greather than specified. Expecting: " + sizeSpec + " Actual: " +   elemValue.length;
        logger.error("Tahoe 16801" + elemLabel  + " size greather than specified. elemValue: " + elemValue + "  Expecting: " + sizeSpec + " Actual: " +   elemValue.length + " msg: " + connectorMessage.getRawData());			
    }
	return elemValue;
}

function formatDate(value,dateFormat)
{
	var s="";
	if ((value=="")||(value.length<8)) return "";
	else
	{
		value=value.substring(0,8);	
		s=new java.text.SimpleDateFormat("MM/dd/yyyy").format(new java.text.SimpleDateFormat(dateFormat).parse(value));
	}
	return s;
	
}


String.prototype.trim = function() {
a = this.replace(/^\s+/, '');
return a.replace(/\s+$/, '');
};

**Reply by jonbartels** [reply]
Please use code formatting when posting to GitHub - https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

The code you posted does not match the code in your original screenshot either.

**Reply by Jaysarun** [reply]
Apologies for that, Is it okay now? The screenshot I posted earlier it is from Channel's Processing Error Tab 


`var sqlUtils =  new Packages.com.quantros.utils.SQLUtils();
var DBMS_URL="train";
var vErrors="";
/**************************** START FUNCTIONS ****************************************/

var checkManFields =  function(arrlabels,arr)
{
	var errorMsgs="";
	if (arr.length!=arrlabels.length)
	{
		errorMsgs+="Tahoe 16801 Labels and columns array must match";
	}
	for(var i=0; i<arr.length; i++)
	{
		if (arr[i]=="")
		{
			errorMsgs+="Tahoe 16801 Mandatory: " + arrlabels[i];	
		}		
	}

	return errorMsgs;
}

function executeUpdate(strSQL,inParamsArray,inParamsTypesArray)
{
	return sqlUtils.executeUpdate(DBMS_URL,strSQL,makeJavaArr(inParamsArray),makeJavaArr(inParamsTypesArray));
}

function executeSingleRowQuery(strSQL,inParamsArray,inParamsTypesArray)
{
	var s=sqlUtils.executeSingleRowQuery(DBMS_URL,strSQL,makeJavaArr(inParamsArray),makeJavaArr(inParamsTypesArray));
	var arr = s.split(",");
	return arr;
}


function executeSingleValQuery(strSQL,inParamsArray,inParamsTypesArray)
{
	return sqlUtils.executeSingleValQuery(DBMS_URL,strSQL,makeJavaArr(inParamsArray),makeJavaArr(inParamsTypesArray));
}

function makeJavaArr(jsArray)
{
	var ae = java.lang.reflect.Array.newInstance(java.lang.String, jsArray.length); 
	 for(var i=0; i<jsArray.length; i++)
	 {
	 	ae[i]=jsArray[i];
	 }
	 return ae;
}

function arrayToString(arr)

{
	var s="";
	for(var i=0; i<arr.length; i++)
	{
		s+= arr[i] +",";
	}
	return s;
}

function checkSize(elemLabel,elemValue,sizeSpec)
{
	if (elemValue.length > sizeSpec)
	{
		vErrors += elemLabel + "Tahoe 16801 Size greather than specified. Expecting: " + sizeSpec + " Actual: " +   elemValue.length;
        logger.error("Tahoe 16801" + elemLabel  + " size greather than specified. elemValue: " + elemValue + "  Expecting: " + sizeSpec + " Actual: " +   elemValue.length + " msg: " + connectorMessage.getRawData());			
    }
	return elemValue;
}

function formatDate(value,dateFormat)
{
	var s="";
	if ((value=="")||(value.length<8)) return "";
	else
	{
		value=value.substring(0,8);	
		s=new java.text.SimpleDateFormat("MM/dd/yyyy").format(new java.text.SimpleDateFormat(dateFormat).parse(value));
	}
	return s;
	
}


String.prototype.trim = function() {
a = this.replace(/^\s+/, '');
return a.replace(/\s+$/, '');
};`

**Reply by Jaysarun** [reply]
Attaching the code file below
[DB Config setup code.txt](https://github.com/nextgenhealthcare/connect/files/13954371/DB.Config.setup.code.txt)

**Reply by pacmano1** [reply]
No, please read https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Reply by pacmano1** [reply]
Your code should look like this, just a snippet of your orginal.   Also format your code.

```JS
function formatDate(value, dateFormat) {
    var s = "";
    if ((value == "") || (value.length < 8)) return "";
    else {
        value = value.substring(0, 8);
        s = new java.text.SimpleDateFormat("MM/dd/yyyy").format(new java.text.SimpleDateFormat(dateFormat).parse(value));
    }
    return s;
}
```

**Reply by pacmano1** [reply]
IMHO, your code makes little sense when you can just use a code template like https://github.com/pacmano1/Mirth-Snippets/blob/main/executeSharedDBStatement.js

**Reply by Jaysarun** [reply]
var sqlUtils = new Packages.com.quantros.utils.SQLUtils();
var DBMS_URL="train";
var vErrors="";
/**************************** START FUNCTIONS ****************************************/
var checkManFields = function(arrlabels,arr)
{
var errorMsgs="";
if (arr.length!=arrlabels.length)
{
errorMsgs+="Tahoe 16801 Labels and columns array must match";
}
for(var i=0; i<arr.length; i++)
{
if (arr[i]=="")
{
errorMsgs+="Tahoe 16801 Mandatory: " + arrlabels[i];
}
}
return errorMsgs;
}
function executeUpdate(strSQL,inParamsArray,inParamsTypesArray)
{
return sqlUtils.executeUpdate(DBMS_URL,strSQL,makeJavaArr(inParamsArray),makeJavaArr(inParamsTypesArray));
}
function executeSingleRowQuery(strSQL,inParamsArray,inParamsTypesArray)
{
var s=sqlUtils.executeSingleRowQuery(DBMS_URL,strSQL,makeJavaArr(inParamsArray),makeJavaArr(inParamsTypesArray));
var arr = s.split(",");
return arr;
}
function executeSingleValQuery(strSQL,inParamsArray,inParamsTypesArray)
{
return sqlUtils.executeSingleValQuery(DBMS_URL,strSQL,makeJavaArr(inParamsArray),makeJavaArr(inParamsTypesArray));
}
function makeJavaArr(jsArray)
{
var ae = java.lang.reflect.Array.newInstance(java.lang.String, jsArray.length);
for(var i=0; i<jsArray.length; i++)
{
ae[i]=jsArray[i];
}
return ae;
}
function arrayToString(arr)
{
var s="";
for(var i=0; i<arr.length; i++)
{
s+= arr[i] +",";
}
return s;
}
function checkSize(elemLabel,elemValue,sizeSpec)
{
if (elemValue.length > sizeSpec)
{
vErrors += elemLabel + "Tahoe 16801 Size greather than specified. Expecting: " + sizeSpec + " Actual: " + elemValue.length;
logger.error("Tahoe 16801" + elemLabel + " size greather than specified. elemValue: " + elemValue + " Expecting: " + sizeSpec + " Actual: " + elemValue.length + " msg: " + connectorMessage.getRawData());
}
return elemValue;
}
function formatDate(value,dateFormat)
{
var s="";
if ((value=="")||(value.length<8)) return "";
else
{
value=value.substring(0,8);
s=new java.text.SimpleDateFormat("MM/dd/yyyy").format(new java.text.SimpleDateFormat(dateFormat).parse(value));
}
return s;
}
String.prototype.trim = function() {
a = this.replace(/^\s+/, '');
return a.replace(/\s+$/, '');
};
